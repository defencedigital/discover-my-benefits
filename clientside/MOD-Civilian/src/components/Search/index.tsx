import React from 'react';
import Select from 'react-select';
import { graphql, useStaticQuery, navigate } from 'gatsby';
import { getAllUsedBenefits } from '@utils/helpers';

const query = graphql`
  query getSearchBenefits {
    squidex {
      queryLandingpageContents {
        id
        flatData {
          categories {
            id
            flatData {
              shortIntro
              name
              slug
              popularBenefits {
                flatData {
                  name
                  slug
                  shortIntro
                  parentCategory {
                    flatData {
                      name
                      slug
                    }
                    id
                  }
                }
                id
              }
              subcategorisedBenefits {
                benefits {
                  flatData {
                    name
                    slug
                    shortIntro
                    parentCategory {
                      flatData {
                        name
                        slug
                      }
                      id
                    }
                  }
                  id
                }
              }
              uncategorisedBenefits {
                id
                flatData {
                  name
                  slug
                  shortIntro
                  parentCategory {
                    id
                    flatData {
                      slug
                      name
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
  }
`;

const getoptions = (benefits: [IBenefits], categories: any[]) => {
  const Benefitdata = benefits.map(benefit => {
    const { name, slug, shortIntro } = benefit.flatData;

    const categorySlug = benefit.flatData.parentCategory[0].flatData.slug;
    const label = shortIntro ? `${name} | ${shortIntro}` : `${name}`;
    return Object.assign({
      value: `/${categorySlug}/${slug}`,
      label,
    });
  });
  const Categorydata = categories.map((category: { flatData: { slug: any; name: any } }, index: any) => {
    const { slug, name } = category.flatData;
    return Object.assign({
      value: `${slug}`,
      label: `${name}`,
    });
  });
  return [...Benefitdata, ...Categorydata];
};

const handleChange = (e: { value: any }) => {
  const { value } = e;
  navigate(value);
};

const Search = () => {
  const { squidex } = useStaticQuery(query);
  const { categories } = squidex.queryLandingpageContents[0].flatData;
  const benefits = getAllUsedBenefits(categories);

  const options = getoptions(benefits, categories);
  return (
    <div className="bg-secondary p-2 h-16 h-full">
      <Select
        className="searchable-dropdown max-w-1110 h-full m-auto"
        classNamePrefix="searchable-dropdown"
        onChange={handleChange}
        options={options}
        isSearchable
        placeholder="Search Benefits &amp; Categories"
      />
    </div>
  );
};

export default Search;
