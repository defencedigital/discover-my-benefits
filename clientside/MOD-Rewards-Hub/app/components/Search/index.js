import PropTypes from 'prop-types';
import React from 'react';
import Select from 'react-select';
import 'react-select/dist/react-select.css';
// import { Button } from 'reactstrap';

class Search extends React.PureComponent {
  // eslint-disable-line react/prefer-stateless-function
  state = {
    value: this.props.value || null,
  };

  getSearchSelect() {
    const { options, placeholder, buttonText } = this.props;

    return (
      <Select
        ref={ref => {
          this.select = ref;
        }}
        onBlurResetsInput={false}
        onSelectResetsInput={false}
        options={options}
        simpleValue
        searchable
        clearable
        name="value"
        id={buttonText}
        placeholder={placeholder}
        value={this.state.value}
        onChange={this.updateValue}
        className="search-select"
      />
    );
  }

  updateValue = value => {
    // comment out to automatically submit
    this.props.onSubmit(value);
    this.setState({
      value,
    });
  };

  submit = e => {
    e.preventDefault();
    this.props.onSubmit(this.state.value);
  };

  render() {
    const { buttonText } = this.props;
    return (
      <form className="search-inline" onSubmit={this.submit}>
        <label htmlFor={buttonText} style={{ display: 'none' }}>
          {buttonText}
        </label>
        {this.getSearchSelect()}
        {/* <Button className="search-btn" type="submit">
          {buttonText}
        </Button> */}
      </form>
    );
  }
}

Search.propTypes = {
  options: PropTypes.array.isRequired,
  onSubmit: PropTypes.func.isRequired,
  placeholder: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([PropTypes.string]),
  buttonText: PropTypes.string.isRequired,
};

export default Search;
