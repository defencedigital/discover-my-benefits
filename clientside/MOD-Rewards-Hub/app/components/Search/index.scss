.search-select {
  width: 100%;
  text-align: left;
  display: inline-block;
  // padding-right: 80px;
  width: 100%;
  line-height: 1.9rem;
  vertical-align: top;
  //padding-right: 100px;
  @include media-breakpoint-down(md) {
    padding: 0 20px;
  }

  .Select-arrow-zone {
    display: none;
  }

  .Select-clear{
    font-size: 24px;
  }
}

.role-panel .search-select {
  //padding-right: 170px;
  @include media-breakpoint-down(md) {
    padding: 0 10px;
  }
}

.search-inline {
  position: relative;
  margin: 0 auto;
  max-width: 750px;
  //height: 44px;
	max-height: 49px;
	// padding-right: 90px;
  @include media-breakpoint-down(md) {
    width: 100%;
		margin-bottom: 20px;
		padding-right: 0;
  }
}

.search-btn {
  position: absolute;
  right: 0;
  height: 45px;
  vertical-align: top;
  margin-left: 20px;
  @include media-breakpoint-down(md) {
    display: block;
    margin: 20px auto;
    position: relative;
  }
}

.search-select .Select-control {
  height: 44px;
  max-width: 100%;
  overflow: hidden;
  padding-left: 40px;

  &:before{
    content: '';
    display: block;
    background: url('/' + $service + '/search.svg') no-repeat;
    width: 40px;
    height: 40px;
    position: absolute;
    left: 10px;
    top: 10px;
  }

}

.search-select .Select-placeholder{
  padding-left: 50px;
}

.search-select .Select-multi-value-wrapper,
.search-select .Select-placeholder,
.search-select .Select--single > .Select-control .Select-value {
  height: 100%;
  line-height: 4.5rem;
  font-size: 1.6rem;
  color: $brand-primary;
  margin-top: 1px;
}

.Select.has-value.Select--single > .Select-control .Select-value .Select-value-label,
.Select.has-value.is-pseudo-focused.Select--single > .Select-control .Select-value .Select-value-label {
  line-height: 4.5rem;
  color: $brand-primary;
}

.search-select .Select-menu-outer {
  position: absolute;
  top: 52px;
  width: calc(100% - 0px);
  right: 80px;

  @include media-breakpoint-down(md) {
    width: 100%;
  }
}

.search-select .Select-input {
  height: 100%;
  width: 100%;
  line-height: 4.5rem;
  font-size: 1.6rem;
  overflow: hidden;
  max-width: 100%;
}

.search-select .Select-input input{
  max-width: 613px;

  @include media-breakpoint-down(md) {
    max-width: 100%;
  }
}

.role-panel-search .search-select .Select-input input{
  max-width: 540px;

  @include media-breakpoint-down(md) {
    max-width: 100%;
  }
}

.Select-menu-outer {
  font-size: 1.6rem;
}

.role-panel-search .search-select .Select-menu-outer {
  width: calc(100% - 0px);
  top: 51px;

  @include media-breakpoint-down(md) {
    width: 100%;
  }

}
