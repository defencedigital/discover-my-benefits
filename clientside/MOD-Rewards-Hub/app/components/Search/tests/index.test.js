import React from 'react';
import { fromJS } from 'immutable';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Search from '../index';

import { option } from '../../../tests/commonMocks';

Enzyme.configure({ adapter: new Adapter() });

const mockedProps = fromJS({
  title: 'title',
  text: 'text',
  buttonText: 'buttonText',
  placeholder: 'placeholder',
  options: [option],
  onSubmit: () => {},
});

describe('Search component', () => {
  it('matches the snapshot', () => {
    const html = shallow(<Search {...mockedProps.toJS()} />);
    expect(html).toMatchSnapshot();
  });
});
