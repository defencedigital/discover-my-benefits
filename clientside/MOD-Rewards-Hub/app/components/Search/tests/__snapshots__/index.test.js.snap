// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component matches the snapshot 1`] = `ShallowWrapper {}`;
